File: aws/cdk/typescript/lib/stacks/secretsmanager-stack.ts
import { NestedStack, SecretValue } from "aws-cdk-lib";
import { Secret } from "aws-cdk-lib/aws-secretsmanager";
import { Construct } from "constructs";
import { SecretsManagerStackProps } from "../interfaces/secretsmanager-interface";

export class SecretsManagerStack extends NestedStack {
  public readonly token: Secret;

  constructor(scope: Construct, id: string, props: SecretsManagerStackProps) {
    super(scope, id, props);

    const { secretName, description, secretValue } = props;

    this.token = new Secret(this, "ControlPlaneSecret", {
      description,
      secretName,
      secretStringValue: SecretValue.unsafePlainText(secretValue)
    });
  }
}

File: aws/cdk/java/src/main/java/com/gatlingenterprise/stacks/SecretsManagerStack.java
package com.gatlingenterprise.stacks;

import com.gatlingenterprise.records.controlPlaneProps.SecretsManagerStackProps;
import software.amazon.awscdk.NestedStack;
import software.amazon.awscdk.SecretValue;
import software.amazon.awscdk.services.secretsmanager.Secret;
import software.constructs.Construct;

public class SecretsManagerStack extends NestedStack {
  public static Secret secretToken;

  public SecretsManagerStack(Construct scope, String id, SecretsManagerStackProps props) {
    super(scope, id, props);

    String secretName = props.secretName();
    String desc = props.desc();
    String secretValue = props.secretValue();

    secretToken =
        Secret.Builder.create(this, "ControlPlaneSecret")
            .description(desc)
            .secretName(secretName)
            .secretStringValue(SecretValue.unsafePlainText(secretValue))
            .build();
  }
}
